libass/ass_drawing.c

```diff
@@ -88,9 +102,12 @@ static size_t add_3_points(const char **str, ASS_DrawingToken **tail, ASS_TokenT
     if (!valid)
         return 0;
 
-    add_node(tail, type, buf[0]);
-    add_node(tail, type, buf[1]);
-    add_node(tail, type, buf[2]);
+    if (!add_node(tail, type, buf[0]) ||
+            !add_node(tail, type, buf[1]) ||
+            !add_node(tail, type, buf[2])) {
```

Comment:
Indentation looks horrible. Maybe the same trick with `valid` would be better?
```c
valid = valid && add_node(tail, type, buf[0]);
valid = valid && add_node(tail, type, buf[1]);
valid = valid && add_node(tail, type, buf[2]);
if (!valid) { ... }
```

Reply:
done